src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageReadmeControl.xaml.cs

```diff
@@ -17,54 +20,110 @@ namespace NuGet.PackageManagement.UI
     {
 #pragma warning disable CS0618 // Type or member is obsolete
         private IMarkdownPreview _markdownPreview;
 #pragma warning restore CS0618 // Type or member is obsolete
+        private bool _isDisposed = false;
+        private CancellationTokenSource _controlDisposingTokenSource;
+        private bool _isBusy = true;
 
         public PackageReadmeControl()
         {
             InitializeComponent();
-            _markdownPreview = MarkdownPreviewSingleton.GetInstance();
-            descriptionMarkdownPreview.Content = _markdownPreview.VisualElement;
+            _controlDisposingTokenSource = new CancellationTokenSource();
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        /// <summary>
+        /// Added to allow the view to display the loading spinner while the markdown is being rendered.
+        /// The markdown rendering occurs on the view so the view should have some control over this property.
+        /// </summary>
+        public bool IsBusy
+        {
+            get => _isBusy || ReadmeViewModel.IsBusy;
+        }
+
+        /// <summary>
+        /// Added to allow the view to hide the README while the markdown is being rendered
+        /// The markdown rendering occurs on the view so the view should have some control over this property.
+        /// </summary>
+        public bool IsReadmeReady
+        {
+            get => !_isBusy && ReadmeViewModel.IsReadmeReady;
         }
 
         public ReadmePreviewViewModel ReadmeViewModel { get => (ReadmePreviewViewModel)DataContext; }
 
-        private void ReadmeViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        public bool Initialize(IEditorOptionsFactoryService options)
         {
-            if (e.PropertyName == nameof(ReadmePreviewViewModel.ReadmeMarkdown))
+            if (options is null)
             {
-                NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(UpdateMarkdownAsync).PostOnFailure(nameof(PackageReadmeControl), nameof(ReadmeViewModel_PropertyChanged));
+                return false;
             }
+            if (_markdownPreview is null)
+            {
+                // This class is marked as obsolete because the api hasn't been finalized, however we want to use IMarkdownPreview to maintain a centralized way of rendering markdown into html.
+#pragma warning disable CS0618 // Type or member is obsolete
+                var previewBuilder = new PreviewBuilder();
+#pragma warning restore CS0618 // Type or member is obsolete
+                previewBuilder.EditorOptions = options.GlobalOptions;
+                _markdownPreview = previewBuilder.Build();
+                descriptionMarkdownPreview.Content = _markdownPreview?.VisualElement;
+            }
+            return true;
         }
 
-        private async Task UpdateMarkdownAsync()
+        private void ReadmeViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
```

Comment:
We talked offline:
Eventually, I'd like to see this entire method go away in favor of setting the ViewModel `IsBusy` property instead.
Consider filing a techDebt issue describing the changes we discussed.